src/vos/tests/csum_extent_tests.c

```diff
@@ -882,3 +882,54 @@ csum_extent_not_chunk_aligned(void **state)
 	free(csum_buf_1);
 	free(csum_read_buf);
 }
+
+void csum_invalid_input_tests(void **state)
+{
+	const uint64_t		data_size = 20;
+	const uint64_t		chunk_size = 8;
+
+	struct extent_key	 extent_key;
+	daos_csum_buf_t		 csum;
+	daos_csum_buf_t		 read_csum;
+	uint8_t			*csum_buf_1 = NULL;
+	uint8_t			*csum_zero_buf = NULL;
+	uint8_t			*csum_read_buf = NULL;
+	uint8_t			*data_buf_1 = allocate_random(data_size);
+	uint8_t			*read_data_buf = calloc(data_size, 1);
```

Comment:
Use the allocation macros D_ALLOC, etc.  and add cmocka asserts for NULL

Reply:
I updated to use D_ALLOC, but I'm not exactly sure where you're thinking the NULL asserts should go.